tgui/packages/tgui/interfaces/Manufacturer/index.tsx

```diff
@@ -0,0 +1,379 @@
+
+    </Stack>
+  );
+};
+
+const MaterialRow = (props, context) => {
+  let { act } = useBackend(context);
+  let { resource, resourceAmt } = props;
+
+  return (
+    <TableRow collapsing>
+      <TableCell collapsing>
+        <Button icon="eject" onClick={() => act("material_eject", { "material": resource })} />
+      </TableCell>
+      <TableCell>
+        <Button icon="add" onClick={() => act("material_swap", { "resource": resource })} />
+      </TableCell>
+      <TableCell header>
+        {resource}
+      </TableCell>
+      <TableCell>
+        {resourceAmt/10}
+      </TableCell>
+    </TableRow>
+  );
+};
+
+const LoadedMaterials = (props, context) => {
+  let { resources } = props;
+  let resourceRows = [];
+  for (let resource in resources) {
+    resourceRows.push(<MaterialRow resource={resource} resourceAmt={resources[resource]} />);
+  }
+  return (
+    <Table>
+      {resourceRows}
+    </Table>
+  );
+};
+
+const CategoryDropdown = (props) => {
+  const { category, blueprints } = props;
+  let buttons = [];
+  for (let i in blueprints) {
+    buttons.push(<BlueprintButton name={i} blueprintData={blueprints[i]} />);
+  }
+
+  return (
+    <Collapsible open title={category}>
+      {buttons}
+    </Collapsible>
+  );
+};
+
+const CardInfo = (props, context) => {
+  const { data, act } = useBackend<ManufacturerData>(context);
+  return (data.card_owner === null || data.card_balance === null) ? (
+    <Flex backgroundColor={backgroundPop} py={1}>
+      <Flex.Item grow>
+        No Card Inserted
+      </Flex.Item>
+      <Flex.Item>
+        <Button icon="add" onClick={() => act("card", { "scan": true })}>Insert Card</Button>
+      </Flex.Item>
+    </Flex>
+  ) : (
+    <Box backgroundColor={backgroundPop} py={1}>
+      Card Owner: {data.card_owner}<br />
+      Current Balance: {data.card_balance}⪽
+      <Button icon="subtract" onClick={() => act("card", { "remove": true })}>Remove Card</Button>
+    </Box>
+  );
+};
+
+const is_set = (bits, bit) => { return bits & (1 << bit); };
+
+export const CollapsibleWireMenu = (props, context) => {
+  const { act, data } = useBackend<ManufacturerData>(context);
+  let wireContent = [];
+  let i = 0;
+  wireContent.push(
+    <>
+      <Flex style={{ "align-items": "center" }}>
+        <Flex.Item style={{ "width": "45%" }}>
+          Panel {data.panel_open ? "open" : "closed"}.
+        </Flex.Item>
+        <Flex.Item grow>
+          <Button
+            icon={data.panel_open ? "warning" : "circle"}
+            color={data.panel_open ? "orange" : "green"}
+            onClick={() => act("toggle_panel")}
+          >
+            Screws {data.panel_open ? "Loose" : "Secure"}
+          </Button>
+        </Flex.Item>
+      </Flex>
+      <Divider />
+    </>
+  );
+  if (data.panel_open)
+  {
+    for (let wire in data.wires)
+    {
+      let cut = is_set(data.wire_bitflags, i);
+      i++;
+      wireContent.push(
+        <Flex textColor={data.wires[wire]} px={2}>
```

Comment:
Is `px` really needed here? Also, text coloring should be on the first `Flex.Item` rather than applying to the whole thing.

Reply:
moved to the first child